pkg/logql/log/ip.go

```diff
@@ -0,0 +1,195 @@
+package log
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"unicode"
+)
+
+// This actually implements the Stage interface to process the line
+type LineIPFiler struct {
+	pat string
+}
+
+func NewLineIpFilter(pattern string) (Stage, error) {
+	return &LineIPFiler{pat: pattern}, nil
+}
+
+func (f *LineIPFiler) Process(line []byte, _ *LabelsBuilder) ([]byte, bool) {
```

Comment:
I'd like to see a benchmark of this. I wonder if the `string(line)` creates a copy or not. If so I think you should avoid it.